daffodil-runtime1/src/main/scala/org/apache/daffodil/dsom/CompiledExpression1.scala

```diff
@@ -212,7 +212,48 @@ class DPathCompileInfo(
   extends ImplementsThrowsSDE with PreSerialization
   with HasSchemaFileLocation {
 
+
+  /**
+   * This "parents" val is a backpointer to all DPathCompileInfo's that
+   * reference this DPathCompileInfo. The problem with this is that when
+   * elements are shared, these backpointers create a highly connected graph
+   * that requires a large stack to serialize using the default java
+   * serialization as it jumps around parents and children. To avoid this large
+   * stack requirement, we make the parents backpointer transient. This
+   * prevents jumping back up to parents during serialization and results in
+   * only needing a stack depth relative to the schema depth. Once all that
+   * serialization is completed and all the DPathCompileInfo's are serialized,
+   * we then manually traverse all the DPathCompileInfo's again and serialize
+   * the parent sequences (via the serailizeParents method). Because all the
+   * DPathCompileInfo's are already serialized, this just serializes the
+   * Sequence objects and the stack depth is again relative to the schema
+   * depth.
+   */
+  @transient
   lazy val parents = parentsArg
```

Comment:
lazy val no longer needed as parentsArg is not a by-name argument. Could just rename the arg to parents.

Reply:
Correct. Will fix.

Reply:
Note that there was some issue with ``@TransientParam`` not working as expected if I just renamed the parameter to ``parents`` and made it a ``val``. Change this  from ``lazy val`` to ``val`` does work as expected though.

Reply:
Looks like val's in a constructor parameter cannot be made transient. Keeping the parentsArg and using a lazy val I think is the normal solution to this kindof thing.